metadata-io/src/main/java/com/linkedin/metadata/timeline/eventgenerator/EntityChangeEventGenerator.java

```diff
@@ -0,0 +1,45 @@
+package com.linkedin.metadata.timeline.eventgenerator;
+
+import com.datahub.authentication.Authentication;
+import com.github.fge.jsonpatch.JsonPatch;
+import com.linkedin.common.AuditStamp;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.data.template.RecordTemplate;
+import com.linkedin.entity.client.EntityClient;
+import com.linkedin.metadata.entity.EntityAspect;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.timeline.data.ChangeCategory;
+import com.linkedin.metadata.timeline.data.ChangeEvent;
+import com.linkedin.metadata.timeline.data.ChangeTransaction;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+
+public abstract class EntityChangeEventGenerator<T extends RecordTemplate> {
+  protected EntityService _entityService;
```

Comment:
minor: any chance we can use _entityClient here?

Entity Client is good in situations where we may need to split out into a separate service. In this case, we might.

Reply:
Sure, can do